TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -1190,57 +1317,67 @@ public virtual void StartWhenConditionClause(List<TypeCobol.Compiler.CodeElement
             }
             Exit();// exit WHEN group
             Enter(new Then(), conditions[0]);// enter THEN
+            Dispatcher.StartWhenConditionClause(conditions);
         }
 
 
         public virtual void EndWhenConditionClause()
         {
             Exit();// exit THEN
+            Dispatcher.EndWhenConditionClause();
         }
 
         public virtual void StartWhenOtherClause(TypeCobol.Compiler.CodeElements.WhenOtherCondition cond)
         {
             Enter(new WhenOther(cond), cond);// enter WHEN OTHER
+            Dispatcher.StartWhenOtherClause(cond);
         }
 
         public virtual void EndWhenOtherClause()
         {
             Exit();// exit WHEN OTHER
+            Dispatcher.EndWhenOtherClause();
         }
 
         public virtual void StartIfStatementWithBody(TypeCobol.Compiler.CodeElements.IfStatement stmt)
         {
             Enter(new If(stmt), stmt);
             Enter(new Then(), stmt);
+            Dispatcher.StartIfStatementWithBody(stmt);
         }
         public virtual void EnterElseClause(TypeCobol.Compiler.CodeElements.ElseCondition clause)
         {
             Exit();// we want ELSE to be child of IF, not THEN, so exit THEN
             Enter(new Else(clause), clause);// ELSE
+            Dispatcher.EnterElseClause(clause);
         }
         public virtual void EndIfStatementWithBody(TypeCobol.Compiler.CodeElements.IfStatementEnd end)
         {
             Exit(); // Exit ELSE (if any) or THEN
             AttachEndIfExists(end);
             // DO NOT Exit() IF node because this will be done in ExitStatement
             Exit();//JCM exit any way ???
+            Dispatcher.EndIfStatementWithBody(end);
         }
 
         public virtual void AddNextSentenceStatement(TypeCobol.Compiler.CodeElements.NextSentenceStatement stmt)
         {
             Enter(new NextSentence(stmt));
             Exit();
+            Dispatcher.AddNextSentenceStatement(stmt);
         }
 
         public virtual void StartInvokeStatementConditional(TypeCobol.Compiler.CodeElements.InvokeStatement stmt)
         {
             Enter(new Invoke(stmt), stmt);
+            Dispatcher.StartInvokeStatementConditional(stmt);
         }
 
         public virtual void EndInvokeStatementConditional(TypeCobol.Compiler.CodeElements.InvokeStatementEnd end)
         {
             AttachEndIfExists(end);
             Exit();
+            Dispatcher.EndInvokeStatementConditional(end);
         }
 
         public virtual void StartJsonGenerateStatementConditional(TypeCobol.Compiler.CodeElements.JsonGenerateStatement stmt)
```

Comment:
Missing Dispatch calls for StartJsonGenerateStatementConditional and EndJsonGenerateStatementConditional